# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1279516213 25200
#      Sun Jul 18 22:10:13 2010 -0700
# Node ID 0306e572c686946231a9370de78405c2a3776a20
# Parent  802d0fd0eaf0cd43f86cc691f710595426d77d4c
Use case insensitive sort in spellcheck scripts.

diff --git a/doc/ChangeLog b/doc/ChangeLog
--- a/doc/ChangeLog
+++ b/doc/ChangeLog
@@ -1,12 +1,18 @@
+2010-07-18  Rik <octave@nomad.inbox5.com>
+
+        * interpreter/doccheck/spellcheck,
+	interpreter/doccheck/spellcheck/add_to_aspell_dict:
+	Use case insensitive sort in spellcheck scripts.
+
 2010-07-18  Rik <octave@nomad.inbox5.com>
 
         * interpreter/doccheck/aspell-octave.en.pws: Add Octave-specific words
-	from stats.texi to private Aspell dictionary
+	from stats.texi to private Aspell dictionary.
 
 2010-07-17  Rik <octave@nomad.inbox5.com>
 
         * interpreter/doccheck: New directory for spelling/grammar scripts.
         * interpreter/doccheck/README: Instructions for using scripts.
         * interpreter/doccheck/spellcheck: Script to spellcheck a Texinfo file.
         * interpreter/doccheck/aspell.conf: GNU Aspell configuration file for
         Octave documentation.
diff --git a/doc/interpreter/doccheck/add_to_aspell_dict b/doc/interpreter/doccheck/add_to_aspell_dict
--- a/doc/interpreter/doccheck/add_to_aspell_dict
+++ b/doc/interpreter/doccheck/add_to_aspell_dict
@@ -15,17 +15,17 @@
 if (@ARGV != 1)
 {
    die "USAGE: add_to_aspell_dict <filename_of_new_words>\n";
 }
 
 $new_words_fname = shift(@ARGV);
 if (!-r $new_words_fname)
 {
-	die "Unable to read input file: $new_words_fname\n";
+   die "Unable to read input file: $new_words_fname\n";
 }
 
 ################################################################################
 # Add new words to a dictionary database
 open (FH, "<$new_words_fname") or die "Unable to open file: $new_words_fname\n";
 while (<FH>) { $dict_db{$_} = 1; }
 close (FH);
 
@@ -34,11 +34,11 @@ open (FH, "<$octdict_fname") or die "Una
 $header = <FH>;
 while (<FH>) { $dict_db{$_} = 1; }
 close (FH);
 
 # Remove old dictionary file and write out new one
 unlink ($octdict_fname) or die "Unable to delete Octave dictionary: $octdict_fname\n";
 open (FH, ">$octdict_fname") or die "Unable to open file for writing: $octdict_fname\n";
 print FH $header;
-print FH sort keys(%dict_db);
+print FH sort { uc($a) cmp uc ($b) } keys(%dict_db);
 close (FH);
 
diff --git a/doc/interpreter/doccheck/spellcheck b/doc/interpreter/doccheck/spellcheck
--- a/doc/interpreter/doccheck/spellcheck
+++ b/doc/interpreter/doccheck/spellcheck
@@ -34,13 +34,13 @@ if ($?)
    die ("Aspell command unsuccesful.  Cannot continue\n");
 }
 
 open (FH, "<$tmp_fname") or die "Unable to open misspelled words file: $tmp_fname\n";
 while (<FH>) { $words{$_} = 1; }
 close (FH);
 
 # Print out sorted and uniquified list of misspelled words
-print sort keys(%words);
+print sort { uc($a) cmp uc ($b) } keys(%words);
 
 ################################################################################
 # Clean up temporary files
 unlink ($tmp_fname) or die "Unable to delete temporary file: $tmp_fname\n";
